Source/Core/TopLevelDecoder.swift

```diff
@@ -0,0 +1,7 @@
+import Foundation
+
+protocol TopLevelDecoder {
```

Comment:
confused on the naming here. What is a `TopLevelDecoder`? In 3DS SDK we just refer to this protocol as `Decoding`

Reply:
Please see: https://developer.apple.com/documentation/combine/topleveldecoder

Personally, I believe this is clearer than `Decoding`, which suggests a relationship with `Decoder`.

Reply:
Ah ok makes sense. @daven-gomes-cko @deepesh-vasthimal-cko do we want to change `Decoding` to `TopLevelDecoder` on 3DS?

Reply:
Yeah I think that's a good idea